elfo-network/src/worker/mod.rs

```diff
@@ -902,6 +902,9 @@ impl remote::RemoteHandle for RemoteHandle {
                 Ok(false) => unreachable!(),
                 Err(_) => {}
             }
+        } else {
+            // Forget token to prevent recursion on drop
+            token.forget();
```

Comment:
I'm sure this should be called unconditionally before `trace!`. And.. the better comment is required

Reply:
Fixed + i've added test to reproduce the issue.